scout/adapter/mongo/query.py

```diff
@@ -404,60 +411,44 @@ def affected_inds_query(self, mongo_query, case_id, gt_query):
         ]:  # Consider situation where all individuals are unaffected
             mongo_query["samples"] = affected_query
 
-    def clinsig_query(self, query, mongo_query):
-        """Add clinsig filter values to the mongo query object
-
-        Args:
-            query(dict): a dictionary of query filters specified by the users
-            mongo_query(dict): the query that is going to be submitted to the database
-
-        Returns:
-            clinsig_query(dict): a dictionary with clinsig key-values
-
-        """
+    def clinsig_query(self, query: dict):
+        """Add clinsig filter values to the mongo query object"""
         LOG.debug("clinsig is a query parameter")
         trusted_revision_level = TRUSTED_REVSTAT_LEVEL
         rank = []
         str_rank = []
-        clnsig_query = {}
 
         for item in query["clinsig"]:
             rank.append(int(item))
             # search for human readable clinsig values in newer cases
             rank.append(CLINSIG_MAP[int(item)])
             str_rank.append(CLINSIG_MAP[int(item)])
 
-        if query.get("clinsig_confident_always_returned") is True:
-            LOG.debug("add CLINSIG filter with trusted_revision_level")
+        elem_match_or = {
+            "$or": [
+                {"value": {"$in": rank}},
+                {"value": re.compile("|".join(str_rank))},
+            ]
+        }
 
+        if query.get("clinsig_confident_always_returned") is True:
             clnsig_query = {
                 "clnsig": {
                     "$elemMatch": {
                         "$and": [
-                            {
-                                "$or": [
-                                    {"value": {"$in": rank}},
-                                    {"value": re.compile("|".join(str_rank))},
-                                ]
-                            },
+                            elem_match_or,
                             {"revstat": re.compile("|".join(trusted_revision_level))},
                         ]
                     }
                 }
             }
         else:
-            LOG.debug("add CLINSIG filter for rank: %s" % ", ".join(str(query["clinsig"])))
+            clnsig_query = {"clnsig": {"$elemMatch": elem_match_or}}
+
+        if query.get("clinvar_tag"):
```

Comment:
This is the fix: this filter is removed from the secondary query and set together with the other 2 clinvar-based filters, in the function dedicated to set the clinvar filters